src/controllers/guests/posts.rs

```diff
@@ -2,44 +2,36 @@ use crate::controllers::admin::posts_controller::number_posts_count;
 use crate::controllers::constants::Configuration;
 use crate::controllers::helpers::pagination_logic::{general_category, index_pagination};
 use crate::model::categories::{all_categories_db, category_db, category_pagination_logic};
-use crate::model::posts::{query_single_post, single_post_db, specific_page_posts};
+use crate::model::posts::{single_post_db, specific_page_posts};
 use actix_http::header::LOCATION;
 use actix_web::http::header::ContentType;
 use actix_web::{web, HttpResponse, Responder};
 use handlebars::Handlebars;
 use serde_json::json;
 
-
 // @desc    Redirect user to index
 // @route   GET /
-// // @access  Public
-// #[post("/my-api")]
-// #[doc("My API description")]
-// #[openapi(description = "My API description for Swagger")]
+// @access  Public
+
 pub async fn redirect_user() -> impl Responder {
     web::Redirect::to("/posts/page/1")
 }
 
-pub const SET_POSTS_PER_PAGE:i64=3;
+pub const SET_POSTS_PER_PAGE: i64 = 3;
 
 pub async fn index(
     params: web::Path<i32>,
     config: web::Data<Configuration>,
     handlebars: web::Data<Handlebars<'_>>,
 ) -> Result<HttpResponse, actix_web::Error> {
     let db = &config.database_connection;
-    let total_posts_length = number_posts_count(db).await?;
+    let total_posts_length = number_posts_count(db).await? + 2;
     let posts_per_page_constant = SET_POSTS_PER_PAGE;
-    let mut posts_per_page = total_posts_length / posts_per_page_constant;
-    let check_remainder = total_posts_length % posts_per_page_constant;
-    if check_remainder != 0 {
-        posts_per_page += 1;
-    }
+    let posts_per_page = total_posts_length / posts_per_page_constant;
     let posts_per_page = posts_per_page as usize;
     let param = params.into_inner();
```

Comment:
`params` -> `param` cause it's only one parameter

Reply:
fixed it